EtherKit/Models/TypedData.swift

```diff
@@ -0,0 +1,402 @@
+//
+//  TypedData.swift
+//  EtherKit
+//
+//  Created by Zac Morris on 2018-06-14.
+//
+import BigInt
+import Foundation
+import Marshal
+
+// Represents the definition of a member variable for a struct type
+public struct MemberVariable: Unmarshaling {
+  var name: String
+  var type: String
+
+  public init(object: MarshaledObject) throws {
+    name = try object.value(for: "name")
+    type = try object.value(for: "type")
+  }
+
+  // Encodes this member variable
+  public func encodeMember() -> String {
+    return "\(type) \(name)"
+  }
+}
+
+// Represents the collection of member variables defining a struct type
+public struct StructTypeDefinition: ValueType {
+  public var members: [MemberVariable]
+
+  public static func value(from object: Any) throws -> StructTypeDefinition {
+    let memberArray: [MemberVariable] = try [MemberVariable].value(from: object)
+    return StructTypeDefinition(members: memberArray)
+  }
+
+  // Returns the collection of struct types used as a member variable by this struct type
+  public func getSubStructTypes() -> [String] {
+    let standardTypes: Set = ["bool", "uint", "int", "address", "string", "bytes"]
+
+    let subTypes: [String] = members.reduce([String](), { arr, prop in
```

Comment:
Use `reduce(into:updateAccumulatingResult:)` instead.  In that function you just mutate the accumulator instead of returning it.